opnode/node/api.go

```diff
@@ -0,0 +1,274 @@
+	}
+	return (*hexutil.Bytes)(b).UnmarshalJSON(text)
+}
+
+func (b *BytesMax32) UnmarshalText(text []byte) error {
+	if len(text) > 64+2 { // account for 0x prefix
+		return fmt.Errorf("input too long, expected at most 32 hex-encoded, 0x-prefixed, bytes: %x", text)
+	}
+	return (*hexutil.Bytes)(b).UnmarshalText(text)
+}
+
+func (b BytesMax32) MarshalText() ([]byte, error) {
+	return (hexutil.Bytes)(b).MarshalText()
+}
+
+func (b BytesMax32) String() string {
+	return hexutil.Encode(b)
+}
+
+type Uint256Quantity = uint256.Int
+
+type Data = hexutil.Bytes
+
+type PayloadID [8]byte
+
+func (b *PayloadID) UnmarshalJSON(text []byte) error {
+	return hexutil.UnmarshalFixedJSON(reflect.TypeOf(b), text, b[:])
+}
+
+func (b *PayloadID) UnmarshalText(text []byte) error {
+	return hexutil.UnmarshalFixedText("PayloadID", text, b[:])
+}
+
+func (b PayloadID) MarshalText() ([]byte, error) {
+	return hexutil.Bytes(b[:]).MarshalText()
+}
+
+func (b PayloadID) String() string {
+	return hexutil.Encode(b[:])
+}
+
+type ExecutionPayload struct {
+	ParentHash    common.Hash     `json:"parentHash"`
+	FeeRecipient  common.Address  `json:"feeRecipient"`
+	StateRoot     Bytes32         `json:"stateRoot"`
+	ReceiptsRoot  Bytes32         `json:"receiptsRoot"`
+	LogsBloom     Bytes256        `json:"logsBloom"`
+	Random        Bytes32         `json:"random"`
+	BlockNumber   Uint64Quantity  `json:"blockNumber"`
+	GasLimit      Uint64Quantity  `json:"gasLimit"`
+	GasUsed       Uint64Quantity  `json:"gasUsed"`
+	Timestamp     Uint64Quantity  `json:"timestamp"`
+	ExtraData     BytesMax32      `json:"extraData"`
+	BaseFeePerGas Uint256Quantity `json:"baseFeePerGas"`
+	BlockHash     common.Hash     `json:"blockHash"`
+	// Array of transaction objects, each object is a byte list (DATA) representing
+	// TransactionType || TransactionPayload or LegacyTransaction as defined in EIP-2718
+	Transactions []Data `json:"transactions"`
+}
+
+type PayloadAttributes struct {
+	// value for the timestamp field of the new payload
+	Timestamp Uint64Quantity `json:"timestamp"`
+	// value for the random field of the new payload
+	Random Bytes32 `json:"random"`
+	// suggested value for the coinbase field of the new payload
+	SuggestedFeeRecipient common.Address `json:"suggestedFeeRecipient"`
+}
+
+type ExecutePayloadStatus string
+
+const (
+	// given payload is valid
+	ExecutionValid ExecutePayloadStatus = "VALID"
+	// given payload is invalid
+	ExecutionInvalid ExecutePayloadStatus = "INVALID"
+	// sync process is in progress
+	ExecutionSyncing ExecutePayloadStatus = "SYNCING"
+)
+
+type ExecutePayloadResult struct {
+	// the result of the payload execution
+	Status ExecutePayloadStatus `json:"status"`
+	// the hash of the most recent valid block in the branch defined by payload and its ancestors
+	LatestValidHash Bytes32 `json:"latestValidHash"`
+	// additional details on the result
+	ValidationError string `json:"validationError"`
+}
+
+type ForkchoiceState struct {
```

Comment:
Did not realize 'forkchoice' was one word ! (presumably the lack of camel case is not an error)

Reply:
See https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#forkchoicestatev1

Engine-API spec regards it as one word. But the consensus specs fork choice doc uses both "fork choice" and "fork-choice": https://github.com/ethereum/consensus-specs/blob/dev/specs/merge/fork-choice.md And other places mix it up as well.

I'll stick with the API naming here.